test/extended/apiserver/disruption.go

```diff
@@ -0,0 +1,496 @@
+			Kind: "ClusterRole",
+			Name: "cluster-admin",
+		},
+		Subjects: []corev1.ObjectReference{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccountName,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	client := oc.AdminAuthorizationClient().AuthorizationV1().ClusterRoleBindings()
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func callRBACHostaccess(ctx context.Context, oc *exutil.CLI, create bool) error {
+	obj := &authv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-privileged", serviceAccountName),
+			Namespace: namespace,
+		},
+		RoleRef: corev1.ObjectReference{
+			Kind: "ClusterRole",
+			Name: "system:openshift:scc:privileged",
+		},
+		Subjects: []corev1.ObjectReference{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccountName,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	client := oc.AdminAuthorizationClient().AuthorizationV1().ClusterRoleBindings()
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func callServiceAccount(ctx context.Context, oc *exutil.CLI, create bool) error {
+	obj := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: namespace,
+		},
+	}
+
+	client := oc.AdminKubeClient().CoreV1().ServiceAccounts(namespace)
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func callProject(ctx context.Context, oc *exutil.CLI, create bool) error {
+	obj := &projv1.Project{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+			Labels: map[string]string{
+				"pod-security.kubernetes.io/enforce": "privileged",
+				"pod-security.kubernetes.io/audit":   "privileged",
+				"pod-security.kubernetes.io/warn":    "privileged",
+			},
+		},
+	}
+
+	client := oc.AsAdmin().ProjectClient().ProjectV1().Projects()
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func fetchFileViaOC(ctx context.Context, nodeName string, filePath string) (string, error) {
+	args := []string{"adm", "node-logs", nodeName, fmt.Sprintf("--path=%s", filePath)}
+	cmd := exec.CommandContext(ctx, "oc", args...)
```

Comment:
Is it possible to use the exiting oc utility? (grep for "node-logs" for examples)

Reply:
Done

Reply:
Not sure why, but this causes the returned logs to be empty. I'll revisit this later

Reply:
Reworked to use kube API